android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -33,14 +35,153 @@ public String getName() {
+                Settings.System.canWrite(getReactApplicationContext()));
+    }
+
+    /**
+     * Invokes the request permission activity to request access for WRITE_SETTINGS.
+     */
+    private void requestSettingsPermission() {
+        Context context = getReactApplicationContext();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.System.canWrite(context)) {
+            Intent intent = new Intent(
+                    Settings.ACTION_MANAGE_WRITE_SETTINGS,
+                    Uri.parse("package:" + context.getPackageName())
+            );
+            mActivity.startActivityForResult(intent, REQUEST_WRITE_SETTINGS_PERMISSION);
+        }
+    }
+
+    /**
+     * Gets the brightness level of the device settings.
+     *
+     * @return The brightness level.
+     */
+    private Integer getSystemBrightness() {
+        Integer brightness;
+        try {
+            brightness = Settings.System.getInt(
+                    getReactApplicationContext().getContentResolver(),
+                    Settings.System.SCREEN_BRIGHTNESS
+            );
+        } catch (Settings.SettingNotFoundException e) {
+            brightness = null;
+        }
+        return brightness;
+    }
+
+    /**
+     * Sets the brightness level to the device settings.
+     *
+     * @param brightness The brightness level between 0-255.
+     * @return True if the brightness has been set.
+     */
+    private boolean setSystemBrightness(int brightness) {
+        if (hasSettingsPermission()) {
+            // ensure brightness is bound between range 0-255
+            brightness = Math.max(BRIGHTNESS_MIN, Math.min(brightness, BRIGHTNESS_MAX));
+            Settings.System.putInt(
+                    getReactApplicationContext().getContentResolver(),
+                    Settings.System.SCREEN_BRIGHTNESS,
+                    brightness
+            );
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Determines if the application has been granted WRITE_SETTINGS permissions.
+     * This method is callable from the JS context.
+     *
+     * @param promise A promise resolving if the application has WRITE_SETTINGS granted
+     */
+    @ReactMethod
+    public void hasPermission(final Promise promise) {
+        promise.resolve(hasSettingsPermission());
+    }
+
+    /**
+     * Invokes the permission request flow.
+     * This method is callable from the JS context.
+     */
+    @ReactMethod
+    public void requestPermission() {
+        requestSettingsPermission();
+    }
+
+    /**
+     * Updates the device brightness.
+     * This method is callable from the JS context.
+     *
+     * @param brightness The brightness level between 0-1.
+     * @param promise A promise resolving if the brightness was updated.
+     */
+    @ReactMethod
+    public void setBrightness(float brightness, final Promise promise) {
+        if (setSystemBrightness((int) brightness * BRIGHTNESS_MAX)) {
```

Comment:
Wait, if we want to set the brightness to half value, we'd call `setBrightness(0.5)`.

If we cast the float of `0.5` to an int, we'd be rounding it (down I think). So this method would be binary - either max or min brightness, not anywhere in the middle